### PR TITLE
Add formatting function to intl section

### PR DESCRIPTION
**Pull request by moritztk**

### CODE REVIEW

File: desk_reference/localization.md [hunk 1]
```diff
@@ -72,6 +72,45 @@ function BirdSighting({ amountBirds }) {
...
+}
+
+export default injectIntl(InternationalGreeting) // adds `intl` ro props
+```
+
+## FAQ
+
+### When should I use `<FormattedMessage />`
+
+Always, unless you run into major, unexpected issues.
```
Comment (ivan-navarro-75) [inline desk_reference/localization.md]: FormattedMessage let's the translator use the in place translation (edit the content directly on our site)

Comment (moritztk) [reply desk_reference/localization.md]: Yes, this why we should use it always. 
This reference (for now), just explains what to do, not (yet) necessarily goes into detail for the "why" (since that would take much more space.

File: desk_reference/localization.md [hunk 2]
```diff
@@ -72,6 +72,45 @@ function BirdSighting({ amountBirds }) {
 }
 ```
 
-# Resources
+## Localization function
+
+- Should hardly ever be used
```
Comment (Gpx) [inline desk_reference/localization.md]: I would rephrase this part, to clarify why you should not use it.

> You should try to use `FormattedMessage` whenever you can because it allows translators to enter new strings directly from the app. However, `FormattedMessage` creates a `span` which is not suitable for every place. Think of the `placeholder` property for `input`. In such cases, you must use a string and to do so you can use `intl.formatMessage`.

File: desk_reference/localization.md [hunk 3]
```diff
@@ -72,6 +72,45 @@ function BirdSighting({ amountBirds }) {
...
+- Only use when you absolutely cannot use `<FormattedMessage />`
+
+```jsx
+import { injectIntl, defineMessages } from 'react-intl'
+
+const messages = defineMessages({
+  greetings: {
+    id: 424801d3-5e93-463c-99e2-2e2b5cd95ded, // unique id created with uuidgen | awk '{print tolower($0)}'
```
Comment (Gpx) [inline desk_reference/localization.md]: There are other ways to generate uuid, maybe we can add a section at the end to explain how. For instance, I just go here https://duckduckgo.com/?q=uuid&ia=answer

File: desk_reference/localization.md [hunk 4]
```diff
@@ -72,6 +72,45 @@ function BirdSighting({ amountBirds }) {
...
+    description: "Greeting on the main page"
+    defaultMessage: 'Hello!',
```
Comment (Gpx) [inline desk_reference/localization.md]: If you use `Hello {name}!` you can show how to pass variables too

File: desk_reference/localization.md [hunk 5]
```diff
@@ -72,6 +72,45 @@ function BirdSighting({ amountBirds }) {
...
+- Should hardly ever be used
+- Only use when you absolutely cannot use `<FormattedMessage />`
+
+```jsx
+import { injectIntl, defineMessages } from 'react-intl'
+
+const messages = defineMessages({
+  greetings: {
+    id: 424801d3-5e93-463c-99e2-2e2b5cd95ded, // unique id created with uuidgen | awk '{print tolower($0)}'
+    description: "Greeting on the main page"
```
Comment (acabelloj) [inline desk_reference/localization.md]: Can we write that description is a `must do`? We cannot create messages without description

File: desk_reference/localization.md [hunk 6]
```diff
@@ -72,6 +72,45 @@ function BirdSighting({ amountBirds }) {
...
+function InternationalGreeting({ intl }) {
+  const localizedGreetingString = intl.formatMessage(messages.greetings)
+  return (
+    <div>
+      {localizedGreetingString}
+    </div>
+  )
+}
+
+export default injectIntl(InternationalGreeting) // adds `intl` ro props
```
Comment (Gpx) [inline desk_reference/localization.md]: Typo

File: desk_reference/localization.md (hunk 1 above)
Comment (Gpx) [inline desk_reference/localization.md]: I would say "unless you need a string without HTML"

Comment (Gpx) [reply desk_reference/localization.md]: I agree with @ivan-navarro-75 on this (see my comment) I think just a couple of lines of explanation put everything in context and make it easier to understand this part

File: desk_reference/localization.md (hunk 5 above)
Comment (Gpx) [reply desk_reference/localization.md]: And then we can put a link in the (soon to be made) lint rule

Comment (moritztk) [reply desk_reference/localization.md]: Already proposed that for the next guild:
https://trello.com/c/khv5B0sD/147-use-description-in-any-intl-message

File: desk_reference/localization.md (hunk 2 above)
Comment (moritztk) [reply desk_reference/localization.md]: I think such an elaborated text would go beyond the scope of a quick reference. Also those long texts tend to be easily outdated. We can state that it makes things easier for the translator, but I wouldn't get into detail at this place (at least not for now). 
We can add it to a sort of deep-dive doc (like we have for testing or writing a component) and link to it from here, though.

Comment (moritztk) [reply desk_reference/localization.md]: Generally I would keep this as lean as possible, and unless the question doesn't come up naturally.

File: desk_reference/localization.md (hunk 3 above)
Comment (moritztk) [reply desk_reference/localization.md]: So this is the way @acabelloj suggested it. 
I frankly think we should find a way to not have to manually generate UUIDs, to keep things simple.

File: desk_reference/localization.md (hunk 1 above)
Comment (moritztk) [reply desk_reference/localization.md]: This is covered in the FAQ.

Comment (moritztk) [reply desk_reference/localization.md]: I think while there is a place for people who are interested in why we made certain choices, this is not it. 

For the MVP this should be the place on how we do things. Once we got this up and running and we find that it is missing something, we can still add it.
